cmd/snap/cmd_run.go

```diff
@@ -954,3 +967,88 @@ func (x *cmdRun) runSnapConfine(info *snap.Info, securityTag, snapApp, hook stri
 		return syscallExec(cmd[0], cmd, env)
 	}
 }
+
+func randomUUID() (string, error) {
+	uuidBytes, err := ioutil.ReadFile("/proc/sys/kernel/random/uuid")
+	return strings.TrimSpace(string(uuidBytes)), err
+}
+
+var createTransientScope = func(securityTag string) error {
+	if !features.RefreshAppAwareness.IsEnabled() {
+		return nil
+	}
+
+	// The scope is created with a DBus call to systemd running either on
+	// system or session bus, depending on if we are starting a program as root
+	// or as a regular user.
+	var conn *dbus.Conn
+	var err error
+	if os.Getuid() == 0 {
+		conn, err = dbus.SystemBus()
+	} else {
+		conn, err = dbus.SessionBus()
+	}
+	if err != nil {
+		return err
+	}
+
+	// The property and auxUnit types are not documented but can be traced from
+	// systemd source code. Systemd defines the signature of StartTransientUnit
+	// as "ssa(sv)a(sa(sv))". The signature can be decomposed as follows:
+	//
+	// unitName string // name of the unit to start
+	// jobMode string  // corresponds to --job-mode= (see systemctl(1) manual page)
+	// properties []struct{
+	//   Name string
+	//   Value interface{}
+	// } // properties describe properties of the started unit
```

Comment:
unrelated but interesting, will this allow us to do set resource-control properties at some point in the future?

Reply:
Yes

We can also use it to reliably terminate hooks, for example.